### ISSUE TITLE
Implement hyperpolarizability interaction

### ISSUE THREAD

**Issue by yachmena** [robochimps/richmol#2]
Implement first-hyperpolarizability interaction between molecule and external field

$$
-\sum_{A,B,C=X,Y,Z}\langle \Psi_\text{rot}' |\beta_{ABC}|\Psi_\text{rot}\rangle E_A(t)E_B(t)E_C(t)
$$

**Comment by yachmena** [comment]
Hi @matillda123, take a look at this, if this is what you need.

**Comment by yachmena** [comment]
Hi @matillda123, I have implemented hyperpolarizability interaction and added an example of two-colour orientation for the NH3 moleucle, see [examples/nh3_two_color.ipynb](https://github.com/robochimps/richmol/blob/main/examples/nh3_two_color.ipynb), for which I happen to have polarizability and  first hyperpolarizability tensors.

Note that this implementation is preliminary, it has not yet been tested or validated against benchmarks or experimental data

**Comment by yachmena** [comment]
Yes, we can use jax for propagating derivatives, which surely can be implemented but will take some effort. The most difficult part will be implementing Chebyshev or Lanczos TDSE solution natively in jax.

**Comment by yachmena** [comment]
Hi @matillda123, we've added an option to select the angular-momentum projection quantum number $m$, which remains a good quantum number when all fields are applied along the $Z$ axis.  
This allows running separate calculations for different $m$ values, making them faster and less memory-intensive.  
For example, in [examples/nh3_two_color.ipynb](https://github.com/robochimps/richmol/blob/main/examples/nh3_two_color.ipynb), add `m_list` parameter to `RotStates` initialisation:
```python
# ...

jmax = 10

# Specify the m quantum number via the `m_list` parameter to RotStates.
# Since the field will be applied along the Z-axis, m remains a good quantum number.
# If omitted, the calculation becomes more expensive, including all values m=-j..j for j=0..jmax.

states = RotStates.from_geometry(
    jmax,
    xyz,
    m_list=[0],
    print_enr=True,
)

# ...
```
